addons/mail/models/mail_thread.py

```diff
@@ -121,15 +121,18 @@ def _compute_message_partner_ids(self):
     def _search_message_partner_ids(self, operator, operand):
         """Search function for message_follower_ids"""
         neg = ''
+        MailFollowers = self.env['mail.followers'].sudo()
+
         if operator in expression.NEGATIVE_TERM_OPERATORS:
             neg = 'not '
             operator = expression.TERM_OPERATORS_NEGATION[operator]
-        followers = self.env['mail.followers'].sudo()._search([
+        followers_query = MailFollowers._where_calc([
             ('res_model', '=', self._name),
             ('partner_id', operator, operand),
         ])
+        MailFollowers._apply_ir_rules(followers_query)
         # use inselect to avoid reading thousands of potentially followed objects
-        return [('id', neg + 'inselect', followers.subselect('res_id'))]
+        return [('id', neg + 'inselect', followers_query.subselect('res_id'))]
```

Comment:
Isn't it simpler to just add 
```
if not followers:
    return [('id', 'in', [])]
```
This way it doesn't even have to do the query as it will get `is_false` again (whereas with a subquery it will still run it)